rust-libretro/src/environment.rs

```diff
@@ -595,7 +645,9 @@ pub unsafe fn get_core_assets_directory<'a>(callback: retro_environment_t) -> Op
 /// files. Cores that need to be backwards-compatible can still check
 /// [`get_system_directory`].
 #[proc::context(GenericContext)]
-pub unsafe fn get_save_directory<'a>(callback: retro_environment_t) -> Option<&'a Path> {
```

Comment:
`RETRO_ENVIRONMENT_GET_SAVE_DIRECTORY` can return `NULL` to indicate that no save data directory is available. This is expected behavior, so I'd make the return type something like `Result<Option<&'a Path>, EnvironmentCallError>`.